api/src/security/auth-utils.test.ts

```diff
@@ -270,7 +278,161 @@ describe('authorize', function () {
       await auth_utils.authorize({ keycloak_token: 'some token' }, ['abc']);
       expect.fail();
     } catch (actualError) {
-      expect(actualError.message).to.contain('Access Denied');
+      expect(actualError).instanceOf(HTTP403);
+    }
+  });
+
+  it('throws HTTP403 when stubbed getSystemUser throws error', async function () {
+    sinon.stub(auth_utils, 'getSystemUser').callsFake(async function (keycloak_token: object) {
+      if (!keycloak_token || keycloak_token) {
+        throw new Error();
+      }
+    });
+
+    try {
+      await auth_utils.authorize({ keycloak_token: 'any token' }, ['abc']);
+      expect.fail();
+    } catch (actualError) {
+      expect(actualError).instanceOf(HTTP403);
+    }
+  });
+
+  it('throws HTTP403 when stubbed getSystemUser returns user with no roles', async function () {
+    sinon.stub(auth_utils, 'getSystemUser').resolves({
+      id: 0,
+      user_identifier: 'somebody',
+      role_ids: [],
+      role_names: []
+    });
+    sinon.stub(auth_utils, 'userHasValidSystemRoles').returns(false);
+
+    try {
+      await auth_utils.authorize({ keycloak_token: 'any token' }, ['abc']);
+      expect.fail();
+    } catch (actualError) {
+      expect(actualError).instanceOf(HTTP403);
+    }
+  });
+
+  it('authorizes a user with valid roles', async function () {
+    sinon.stub(auth_utils, 'getSystemUser').resolves({
+      id: 0,
+      user_identifier: 'somebody',
+      role_ids: [],
+      role_names: []
+    });
+
+    try {
+      const result = await auth_utils.authorize({ keycloak_token: 'any token' }, ['abc']);
+      expect(result).to.be.true;
+    } catch {
+      expect.fail();
```

Comment:
shouldnt the expect fail be in the try and the catch needs to actually assert the error?

Reply:
I think this test is asserting happy path, so we dont expect it to fail.